src/catalog/src/builtin.rs

```diff
@@ -2899,7 +2941,8 @@ WHERE mz_schemas.database_id IS NULL OR d.name = pg_catalog.current_database()",
 pub const PG_DEPEND: BuiltinView = BuiltinView {
     name: "pg_depend",
     schema: PG_CATALOG_SCHEMA,
-    sql: "CREATE VIEW pg_catalog.pg_depend AS
```

Comment:
This relates to why the tests fail: 

The sql below doesn't properly transform:

expected: 
`mz_internal.mz_type_pg_metadata.typinput::pg_catalog.regproc AS typinput,`

but instead:
`(CASE mztype WHEN 'a' THEN 'array_in' ELSE NULL END)::pg_catalog.regproc AS typinput,`

I'm a bit confused to why this happens, as you can see, I don't touch anything other than skimming the top off.

Reply:
[LINE 935](https://www.diffchecker.com/BXzzhRQ4/)

Reply:
Hmm, weird. Line 935 in that diff is for `pg_type`, but the error and this view is `pg_depend`. I'm not sure I understand the connection yet and why there is a difference. It's also interesting that this only happens in the legacy upgrade tests. It implies the problem only happens after/during an upgrade.